autogl/solver/utils.py

```diff
@@ -122,34 +154,67 @@ def get_best_model(self, index=0) -> str:
         name: `str`
             The name/identifier of the required model.
         """
-        sorted_df = self.perform_dict.sort_values(
-            by=self.major_field, ascending=not self.is_higher_better[self.major_field]
+        sorted_performance_df = self.__performance_data_frame.sort_values(
+            self.__major_field,
+            ascending=not (
+                dict(zip(self.__fields, self.__is_higher_better))[self.__major_field]
+                if self.__major_field in self.__fields else True
+            )
         )
-        name_list = sorted_df["name"].tolist()
+        name_list = sorted_performance_df["name"].tolist()
         if "ensemble" in name_list:
             name_list.remove("ensemble")
         return name_list[index]
 
-    def show(self, top_k=-1) -> None:
+    def show(self, top_k=0) -> None:
         """
-        Show current leaderboard (from good model to bad).
+        Show current LeaderBoard (from best model to worst).
 
         Parameters
         ----------
         top_k: `int`
-            Controls the number model shown. If below `0`, will show all the models. Default `-1`.
+            Controls the number model shown.
+            If less than or equal to `0`, will show all the models. Default to `0`.
 
         Returns
         -------
         None
         """
-        if top_k == -1:
-            top_k = len(self.perform_dict["name"])
+        top_k: int = top_k if top_k > 0 else len(self.__performance_data_frame)
+
+        '''
+        reindex self.__performance_data_frame
+        to ensure the columns of name and representation are in left-side of the data frame
+        '''
+        _columns = self.__performance_data_frame.columns.tolist()
+        maxcolwidths: _typing.List[_typing.Optional[int]] = []
+        if "representation" in _columns:
+            _columns.remove("representation")
+            _columns.insert(0, "representation")
+            maxcolwidths.append(40)
```

Comment:
I _recommend_ the `maxcolwidths` be set as a controllable attribute for this object class, or an argument for this method.